ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -685,6 +685,9 @@ ledgerDbAnchor =
 -- | Information about the state of the most recently flushed ledger
 --
 -- This is what will be serialized when snapshotting.
+--
+-- PRECONDITION: if you are running the legacy ledger, then you must flush
+-- before calling this function
```

Comment:
Maybe we should explain why do we require this precondition.

Reply:
Add this (or some rewording of it) to the haddock above